frontend/src/components/TradingViewWidget.jsx

```diff
@@ -0,0 +1,139 @@
+// TradingViewWidget.jsx
+import React, { useEffect, useRef, memo, useState } from "react";
+import { ScrollArea } from "../components/ui/scroll-area";
+import { Separator } from "../components/ui/separator";
+import { BASE_URL } from "../lib/utils";
+import { UserFullName } from "../context/UserContext";
+
+const TradingViewScrollArea = ({ info }) => {
+  const [allDocuments, setAllDocuments] = useState([]);
+  const { selectedId } = UserFullName();
```

Comment:
Why is `selectedId` (presumably a document ID)` is coming from `UserContext`? Make context name more generic if you store data other than user info. Also, you can define multiple context providers.